enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+- Ensure that any failed PO installations result in a failed cluster installation.
+- Ensure that only AllNamespace OLM-based operators can be installed.
+- Respect any Kubernetes/OCP cluster version constraints while selecting candidate bundles.
+- Respect the "operatorframework.io/suggested-namespace" CSV annotation value.
+
+### Non-Goals
+
+- Prohibit multiple PO packages from being installed on the cluster during phase 0.
+- Provide a first-class UX for configuring POs via the install-config API during phase 0.
+- Provide a unified admin UX for installing OLM-based operators through this PO mechanism vs. the normal OLM workflow during phase 0.
+- Enable POs to block cluster upgrades during phase 0.
+- Onboard optional, CVO-based operators to OLM's management during phase 0.
+- Detach optional, CVO-based operators from the [payload lifecycle](#payload-lifecycle) during phase 0.
+- Enable all OLM-based operators to be deployed via the PO mechanism during phase 0.
+- Support any essential operator to be deployed via the PO mechanism. Future phase designs may re-evaluate this agent-based approach.
+- Support for extending the PO mechanism to cover any MAPI/CCM use cases that will enable supplementing OpenShift with platform-specific integrations during phase 0.
+- Support sourcing catalog content from any CatalogSource in the cluster during phase 0.
+- Support dependency resolution for PO packages during phase 0.
+- Support automatic upgrades for PO packages during phase 0.
+- Support more fine-tuned health checks during phase 0.
+- Require PO packages to report state through an auxiliary API (e.g. ClusterOperator, OperatorCondition, etc.) during phase 0.
+- Provide a mechanism for admins to manually approve PO package upgrades during phase 0.
+- Permit non-essential operators to be rolled back.
+
+## Terminology
+
+### Payload Lifecycle
+
+These are components that build/ship with the OCP payload as a unit. This unit includes both core, control plane components, in addition to the independent, "value" add components that all release on the same release cadence.
+
+### Cluster Lifecycle
+
+These are components that are automatically upgraded when the cluster gets upgraded, and can block cluster lifecycle events:
+
+- Block cluster rollout when an individual component cannot successfully install
+- Block cluster upgrades when an individual component cannot tolerate the incoming version, or has another reason to block the upgrade.
+
+These are the components that install and upgrade with the OCP payload, and influence cluster lifecycle events noted above.
+
+### Non-Essential Operators
+
+These are payload components that don't contribute to a working OpenShift control plane.
+
+### Plain Bundle Format
+
+A "plain" bundle format is OLM's newest bundle format that supports packaging arbitrary Kubernetes resources. This bundle format can also be referred to as the plain+v0 bundle format.
```

Comment:
I realize this entire EP is being viewed in the tech preview mindset, but should we explicitly call out that this format is still considered non-stable and that it may change prior to GA of rukpak (and therefore also prior to GA of PO)?

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/bc14b4537125d43153cc452cd686e02cf96630b2.